fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/client/item/CustomModeledArmorRegistryImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.item;
+
+import java.util.Map;
+
+import it.unimi.dsi.fastutil.objects.Reference2ObjectOpenHashMap;
+
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.client.item.v1.CustomModeledArmorProvider;
+import net.fabricmc.fabric.api.client.item.v1.CustomModeledArmorRegistry;
+
+public final class CustomModeledArmorRegistryImpl implements CustomModeledArmorRegistry {
+	public static final CustomModeledArmorRegistry INSTANCE = new CustomModeledArmorRegistryImpl();
+	private final Map<Item, CustomModeledArmorProvider> providerMap = new Reference2ObjectOpenHashMap<>();
```

Comment:
I think we should mixin to item and store the model provided on items to reduce `map.get` calls